pkg/services/ngalert/state/manager.go

```diff
@@ -30,6 +29,11 @@ type AlertInstanceManager interface {
 	GetStatesForRuleUID(orgID int64, alertRuleUID string) []*State
 }
 
+type StatePersister interface {
+	Async(ctx context.Context, ticker *clock.Ticker, cache *cache)
+	Sync(ctx context.Context, span trace.Span, states, staleStates []StateTransition)
```

Comment:
Would it be useful to also return the error?

```suggestion
	Sync(ctx context.Context, span trace.Span, states, staleStates ([]StateTransition, error)
```

Reply:
No, as we don't do anything with it as logging it, which already happens within the sync call for now.